### PR TITLE
Changes needed for BEAVRS script with domain decomposition and distributed materials/tallies

### PR DESCRIPTION
**Pull request by nhorelik**
This PR adds a few minor odds and ends used with the BEAVRS distributed-material, DD generator script:
- Adds n2n, n3n, and n4n reaction types
- ~~Adds support for scatter and MT tally score types (was this in there previously, and removed for some reason?)~~
- Fixes a ~~few~~ minor bug~~s~~ with materials
- Adds support for the count interactions DD settings feature
- Adds an optimization during XML cell writing (the old way was very slow for large geometries)

### CODE REVIEW

File: src/utils/openmc/universe.py [hunk 1]
```diff
@@ -15,6 +15,8 @@
 # A static variable for auto-generated Cell IDs
 AUTO_CELL_ID = 10000
 
+WRITTEN_IDS = {}
```
Comment (wbinventor) [inline src/utils/openmc/universe.py]: Add a brief comment above this specifying the need for ``WRITTEN_IDS``

File: src/utils/openmc/material.py [hunk 2]
```diff
@@ -76,7 +76,9 @@ def set_id(self, material_id=None):
 
         # If the Material already has an ID, remove it from global list
         if not self._id is None:
-            MATERIAL_IDS.remove(self._id)
+            try:
```
Comment (wbinventor) [inline src/utils/openmc/material.py]: Perhaps this ``try-catch`` block should be replicated for other class' ``set_id(...)`` routines for uniformity? Perhaps ``Universe``, ``Cell``, ``Lattice``, etc.

**Comment by wbinventor** [comment]
Thanks for this PR @nhorelik - I just made two small comments and will merge this in once they have been addressed.

File: src/utils/openmc/material.py (hunk 2 above)
Comment (nhorelik) [inline src/utils/openmc/material.py]: Okay.  Before I do that though, do you agree with what what it's doing here?

Comment (wbinventor) [inline src/utils/openmc/material.py]: Ok, if I understand it correctly, your modification allows the user to set the ``_id`` attribute without using the ``set_id(...)`` routine. In this scenario, the global ``MATERIAL_IDS`` list will not contain the ``Material's`` id, hence the need for the ``try-catch``. Or is there some other scenario you are trying to accommodate?

Comment (nhorelik) [inline src/utils/openmc/material.py]: If i remember correctly that is what I had been doing at one point, but now that I look through the cleaned-up BEAVRS scripts again it seems we no longer are doing that (and shouldn't be - set_id should always be used, I beleive).  I'm thinking I'll actually remove this change.

Comment (wbinventor) [inline src/utils/openmc/material.py]: Cool - if you don't need this change that that's would be preferable.

One other question for you - have you tested this out with the latest
version of OpenCG? Just make sure to update your fork before building
BEAVRS.

Comment (nhorelik) [inline src/utils/openmc/material.py]: When testing before creating this PR I used a fresh clone of the default branch of https://github.com/mit-crpg/OpenCG.

**Pull request merged**